packages/apputils-extension/src/index.ts

```diff
@@ -531,6 +619,38 @@ namespace Private {
       debouncer = window.setTimeout(() => {
         recover(fn);
       }, SPLASH_RECOVER_TIMEOUT);
+    }).catch(() => { /* no-op */ });
+  }
+
+  /**
+   * Allows the user to clear state if splash screen takes too long.
+   */
+  export
+  function redirect(router: IRouter, warn = false): Promise<void> {
+    const form = createRedirectForm(warn);
+    const dialog = new Dialog({
+      title: 'We have a problem!',
+      body: form,
+      focusNodeSelector: 'input',
+      buttons: [Dialog.okButton({ label: 'Create Workspace' })]
+    });
+
+    return dialog.launch().then(result => {
+      dialog.dispose();
+
+      if (result.value) {
+        const url = `workspaces/${result.value}`;
+
+        // Navigate to a new workspace URL and abandon this session altogether.
+        router.navigate(url, { hard: true, silent: true });
+
+        // This promise will never resolve because the application navigates
+        // away to a new location. It only exists to satisfy the return type
+        // of the `redirect` function.
+        return new Promise<void>(() => { /* no-op */ });
```

Comment:
Any difference between this and returning `Promise.resolve()`?

Reply:
Nope. I think in the moment I must have forgotten about `Promise.resolve()` 😄

Reply:
Right. I want a promise that will never resolve. Not a resolve-immediately promise.

Reply:
Do you want that so that listeners to that promise are never triggered? I thought that navigating the page away from this URL would already accomplish that.

Reply:
Navigation away *probably* accomplishes that, but that operation is asynchronous and I wouldn't want to return control flow to the `.then()` that follows the `redirect()` call, so instead I want it to basically hang there as the page unloads and navigates away.

Reply:
My use of the word "hang" is imprecise: I don't mean hang synchronously, I just mean the redirect never returns so the state database is never instantiated so the rest of the app, which depends on the state database, never loads.

Reply:
Ah, got it, thanks.